File: 250212/Run Length 인코딩/run-length-encoding.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        StringBuilder sb = new StringBuilder();

        String str = br.readLine();
        int idx = 0;
        int n = str.length();

        while (idx < n) {
            char current = str.charAt(idx);
            int count = 0;
            while (idx < n && str.charAt(idx) == current) {
                idx++;
                count++;
            }
            sb.append(current).append(String.valueOf(count));
        }
        bw.append(String.valueOf(sb.toString().length())).append("\n").append(sb.toString());
        bw.close();
    }
}

File: 250212/Run Length 인코딩/run-length-encoding.py
import sys

input = sys.stdin.readline

string = input().strip()

idx = 0
n = len(string)
result = ""
while(idx < n):
    current = string[idx]
    cnt = 0
    while (idx < n and current == string[idx]):
        idx = idx + 1
        cnt = cnt + 1
    result = result + current + str(cnt)

print(len(result))
print(result)
